### PR TITLE
Search backend: minor tidying

### PR DESCRIPTION
**Pull request by camdencheek**
Just a cleanup pass of the things that don't fit in other PRs. Details inline.

## Test plan

Semantics-preserving and minimal changes.

### CODE REVIEW

File: internal/search/run/repo_has_file.go [hunk 1]
```diff
@@ -83,19 +83,12 @@ func (s *RepoSearch) reposContainingPath(ctx context.Context, clients job.Runtim
 			return nil, err
 		}
 
-		zoektArgs := search.ZoektParameters{
-			Query:          zoektQuery,
-			Typ:            typ,
-			FileMatchLimit: p.FileMatchLimit,
-			Select:         p.Select,
-		}
```
Comment (camdencheek) [inline internal/search/run/repo_has_file.go:91]: This struct was only constructed in order to access its fields for constructing `ZoektRepoSubsetSearch` below. We can just use the values directly.

File: internal/search/zoekt/indexed_search.go [hunk 2]
```diff
@@ -249,9 +249,6 @@ func DoZoektSearchGlobal(ctx context.Context, client zoekt.Streamer, args *searc
 		return nil
 	}
 
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
```
Comment (camdencheek) [inline internal/search/zoekt/indexed_search.go:253]: This and below are unused context wrappers. There is no need to manually cancel a context unless the child context should be canceled before the parent context.

File: internal/search/zoekt/indexed_search.go [hunk 3]
```diff
@@ -581,11 +575,7 @@ func (t *GlobalSearch) Run(ctx context.Context, clients job.RuntimeClients, stre
 	t.GlobalZoektQuery.ApplyPrivateFilter(userPrivateRepos)
 	t.ZoektArgs.Query = t.GlobalZoektQuery.Generate()
 
-	g, ctx := errgroup.WithContext(ctx)
-	g.Go(func() error {
-		return DoZoektSearchGlobal(ctx, clients.Zoekt, t.ZoektArgs, stream)
-	})
-	return nil, g.Wait()
+	return nil, DoZoektSearchGlobal(ctx, clients.Zoekt, t.ZoektArgs, stream)
```
Comment (camdencheek) [inline internal/search/zoekt/indexed_search.go:578]: This errorgroup only ever had one concurrent goroutine, so we can just run that function directly

**Review by rvantonder — APPROVED** [review]
very nice <3

**Pull request merged**